src/components/NewDatePicker/index.native.js

```diff
@@ -14,28 +13,38 @@ import withNavigation from '../withNavigation';
 
 const datepickerPropTypes = {
     ...propTypes,
-    ...withLocalizePropTypes,
     ...keyboardStatePropTypes,
 };
 
+const datePickerDefaultProps = {
+    ...defaultProps,
+};
+
 class NewDatePicker extends React.Component {
     constructor(props) {
         super(props);
 
         this.state = {
             isPickerVisible: false,
-            selectedDate: props.value || props.defaultValue ? moment(props.value || props.defaultValue).toDate() : new Date(),
+            selectedDate: moment(props.value || props.defaultValue).toDate(),
         };
 
         this.showPicker = this.showPicker.bind(this);
         this.reset = this.reset.bind(this);
         this.hidePicker = this.hidePicker.bind(this);
-        this.updateLocalDate = this.updateLocalDate.bind(this);
-
-        this.wrapperRef = React.createRef();
+        this.setDate = this.setDate.bind(this);
+    }
 
-        this.minDate = props.minDate ? moment(props.minDate).toDate() : null;
-        this.maxDate = props.maxDate ? moment(props.maxDate).toDate() : null;
+    /**
+     * Trigger the `onInputChange` handler when the user input has a complete date or is cleared
+     * @param {Date} selectedDate
+     */
+    setDate(selectedDate) {
+        this.setState(() => {
+            this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
+            return {selectedDate};
+        });
+        this.hidePicker();
```

Comment:
```suggestion
        this.setState({selectedDate}, () => {
            this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
            this.hidePicker();
        });
```

Reply:
@s77rt How is it supposed to work? the selected date comes from currentDateView